compiler/suggest.nim

```diff
@@ -463,11 +468,14 @@ when defined(nimsuggest):
       let x = if info == s.info and info.col == s.info.col: ideDef else: ideUse
       suggestResult(conf, symToSuggest(conf, s, isLocal=false, x, info, 100, PrefixMatch.None, false, 0))
 
-proc findDefinition(conf: ConfigRef; info: TLineInfo; s: PSym) =
+proc findDefinition(conf: ConfigRef; info: TLineInfo; s: PSym; usageSym: var PSym) =
   if s.isNil: return
-  if isTracked(info, conf.m.trackPos, s.name.s.len):
-    suggestResult(conf, symToSuggest(conf, s, isLocal=false, ideDef, info, 100, PrefixMatch.None, false, 0))
-    suggestQuit()
+  if isTracked(info, conf.m.trackPos, s.name.s.len) or s == usageSym:
+    if sfForward notin s.flags:
+      suggestResult(conf, symToSuggest(conf, s, isLocal=false, ideDef, info, 100, PrefixMatch.None, false, 0, useSuppliedInfo = s == usageSym))
+      suggestQuit()
+    else:
+      usageSym = s
```

Comment:
I think `if sfForward notin s.flags: suggestQuit()` would be a preferable way to solve this, because then we get both defs

Reply:
My concerns are with 3rd parties editor plugins. I probably can just issue a changelog for this and everyone should be happy though.